src/lightning_app/components/auto_scaler.py

```diff
@@ -0,0 +1,563 @@
+
+        fastapi_app = _create_fastapi("Load Balancer")
+        security = HTTPBasic()
+        fastapi_app.global_request_count = 0
+        fastapi_app.num_current_requests = 0
+        fastapi_app.last_process_time = 0
+        fastapi_app.SEND_TASK = None
+
+        @fastapi_app.on_event("startup")
+        async def startup_event():
+            fastapi_app.SEND_TASK = asyncio.create_task(self.consumer())
+            self._server_ready = True
+
+        @fastapi_app.on_event("shutdown")
+        def shutdown_event():
+            fastapi_app.SEND_TASK.cancel()
+            self._server_ready = False
+
+        def authenticate_private_endpoint(credentials: HTTPBasicCredentials = Depends(security)):
+            AUTO_SCALER_AUTH_PASSWORD = os.environ.get("AUTO_SCALER_AUTH_PASSWORD", "")
+            if len(AUTO_SCALER_AUTH_PASSWORD) == 0:
+                logging.warning("You have not set password for private endpoints!")
+            current_password_bytes = credentials.password.encode("utf8")
+            is_correct_password = secrets.compare_digest(
+                current_password_bytes, AUTO_SCALER_AUTH_PASSWORD.encode("utf8")
+            )
+            if not is_correct_password:
+                raise HTTPException(
+                    status_code=401,
+                    detail="Incorrect password",
+                    headers={"WWW-Authenticate": "Basic"},
+                )
+            return True
+
+        @fastapi_app.get("/system/info", response_model=_SysInfo)
+        async def sys_info(authenticated: bool = Depends(authenticate_private_endpoint)):
+            return _SysInfo(
+                num_workers=len(self.servers),
+                servers=self.servers,
+                num_requests=fastapi_app.num_current_requests,
+                process_time=fastapi_app.last_process_time,
+                global_request_count=fastapi_app.global_request_count,
+            )
+
+        @fastapi_app.put("/system/update-servers")
+        async def update_servers(servers: List[str], authenticated: bool = Depends(authenticate_private_endpoint)):
+            self.servers = servers
+            self._ITER = cycle(self.servers)
+
+        @fastapi_app.post("/api/predict", response_model=OUTPUT_SCHEMA)
+        async def balance_api(inputs: INPUT_SCHEMA):
+            return await self.process_request(inputs)
+
+        uvicorn.run(
+            fastapi_app,
+            host=self.host,
+            port=self.port,
+            loop="uvloop",
+            timeout_keep_alive=self._timeout_keep_alive,
+            access_log=False,
+        )
+
+    def update_servers(self, server_works: List[LightningWork]):
+        """Updates works that load balancer distributes requests to.
+
+        AutoScaler uses this method to increase/decrease the number of works.
+        """
+        old_servers = set(self.servers)
+        self.servers: List[str] = [server.url for server in server_works if server.url]
```

Comment:
`self.servers` update has to be removed since we can't update the work state from a Flow. Muse also needs to be updated.